preproccesing.py

```diff
@@ -0,0 +1,153 @@
+import json
+import os
+import argparse
+import cv2
+import numpy as np
+import open3d
+import configparser
+from imageio import imwrite
+
+
+def undistort(path, camera_matrix, k_vector, p_vector):
+    image = cv2.imread(path, -1)
+    k1, k2, k3, k4, k5, k6 = k_vector
+    p1, p2 = p_vector
+    undistorted = cv2.undistort(
+        image, camera_matrix, np.array([k1, k2, p1, p2, k3, k4, k5, k6])
+    )
+    return undistorted
+
+
+def depth_images_preprocessing(
+    depth_images_path,
+    depth_matrix,
+    k_vector,
+    p_vector,
+    width,
+    height,
+    width_color,
+    height_color,
+    transform_matrix,
+    color_matrix,
+):
+    depth_images = os.listdir(depth_images_path)
+    new_dir = os.path.join(os.path.dirname(depth_images_path), "depth_preprocessed")
+    if not os.path.isdir(new_dir):
+        os.mkdir(new_dir)
+    color_matrix = np.array(color_matrix)
+    color_matrix = np.concatenate((color_matrix, np.zeros((3, 1))), axis=1)
+    color_matrix = np.concatenate((color_matrix, np.array([[0, 0, 0, 1]])), axis=0)
+    intrinsic = open3d.camera.PinholeCameraIntrinsic()
+    intrinsic.width = width
+    intrinsic.height = height
+    intrinsic.intrinsic_matrix = depth_matrix
+    pc = open3d.geometry.PointCloud()
+    transformation_matrix = np.array(transform_matrix)
+    for image in depth_images:
+        # Depth undistortion
+        depth_undistorted = undistort(
+            os.path.join(depth_images_path, image),
+            np.array(depth_matrix),
+            k_vector,
+            p_vector,
+        )
+
+        # Point cloud creating
+        picture = open3d.geometry.Image(depth_undistorted)
+        point_cloud = pc.create_from_depth_image(picture, intrinsic)
+
+        # Point cloud transformation
+        point_cloud = point_cloud.transform(transformation_matrix)
+
+        # Point cloud to depth image
+        points = np.concatenate(
+            (np.asarray(point_cloud.points), np.full((len(point_cloud.points), 1), 1)),
+            axis=1,
+        )
+        result = np.matmul(color_matrix, np.matrix.transpose(points))
+        img = np.zeros((height_color, width_color)).astype(np.uint16)
+        for point in np.matrix.transpose(result):
+            uz, vz, z, _ = point
+            u = round(uz / z)
+            v = round(vz / z)
+            try:
+                img[v, u] = z * 1000
+            except IndexError:
+                pass
+        imwrite(os.path.join(new_dir, image), img)
```

Comment:
Let's use the next style `imageio.imwrite()`, otherwise it looks like function that is implemented by yourself